internal/rbd/controllerserver.go

```diff
@@ -94,25 +94,39 @@ func (cs *ControllerServer) validateVolumeReq(ctx context.Context, req *csi.Crea
 	return nil
 }
 
+// isMultiWriterBlock validates the volume capability slice against the access modes and access type.
+// if the capability is of multi write the first return value will be set to true and if the request
+// is of type block, the second return value will be set to true. Otherwise, false values are returned.
+func isMultiWriterBlock(caps []*csi.VolumeCapability) (bool, bool) {
+	var multiWriter bool
+	var Block bool
+	for _, cap := range caps {
+		if cap.AccessMode != nil {
+			// we check the capability access mode is multi writer regardless of it is single or multi node.
+			if (cap.AccessMode.Mode == csi.VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER) ||
+				(cap.AccessMode.Mode == csi.VolumeCapability_AccessMode_SINGLE_NODE_MULTI_WRITER) {
```

Comment:
> some discussions happened on the same and we can expect data corruption if different apps are operating on the same file/dir in rwo scenario
>>please provide a link to this one so that helps in understanding more about it.

This was discussed in CSI sync up call  (looks like you were absent ) where along with other folks who were in,  we had a long discussion where @pkalever and @nixpanic extensively talk about these scenarios.

Reply:
>we had a long discussion where @pkalever and @nixpanic extensively talk about these scenarios

@humblec Thanks for the details. If it was an extensive long discussion i assume all the technical aspects are discussed and we are completely settled on the things. If yes, I would expect this summary to be captured in a Github issue or PR   for the people who are not part of the meeting and for upstream reference.

Reply:
Summary is the earlier comment https://github.com/ceph/ceph-csi/pull/2359#discussion_r739009748

Reply:
The comment does not provide complete insights (which is not useful at least for me) am more looking for discussion details on data corruption and when it can happen? current we are supporting RWO for a long time I haven't seen any data corruption.

Reply:
I guess the discussion on the other day of more on the RWX rather than on RWO, more specifically about the filesystem volume mode. At the end of the discussions, we had two opinions about allowing {RWX + SINGLE NODE + FS} case

1. We must continue to restrict, MULTINODE MULTIWRITERS for FS Mode. But allow the same for SINGLENODE MULTIWRITERS with some docs and options to enable IMHO (default: disable)
2. @nixpanic concurs with the above but he suggested we don't need any documentation to allow RWX on a single node as this case looks identical to RWO (Niels can still correct me here)

I guess we take it for granted that every admin will be aware of corruption involved in accessing the same file from multiple apps/pods when application-level fencing doesn't exist.